magpie/api/requests.py

```diff
@@ -379,3 +381,27 @@ def get_query_param(request, case_insensitive_key, default=None):
             if param.lower() == key.lower():
                 return request.params.get(param)
     return default
+
+
+def check_network_mode_enabled(view_func):
+    # type: (Callable) -> Callable
```

Comment:
Can you transfer the definitions from https://github.com/crim-ca/weaver/blob/cba028a11d1bee42d5b7c81211ad6ec8b1a717bf/weaver/typedefs.py#L377-L386

and apply them along https://github.com/Ouranosinc/Magpie/blob/b589ab11751e58e97d55c0affb7661b1fb9d5b00/magpie/typedefs.py#L51-L55

to reuse `AnyViewCallable` here

Reply:
Why transfer them from weaver? 

Can't I just do:

```py
from magpie.typedefs import AnyAccessPrincipalType, AnyRequestType
```

and then change it to:

```
    # type: (AnyRequestType) -> AnyRequestType
```

Reply:
In this case, inputs/outputs are not the requests themselves, but a view callable taking a request (and optionally a context), returning a modified one. I have developed this more on Weaver end, and haven't transferred them back in Magpie. (by transfer, I mean copy-paste literally, and then do `from magpie.typedefs import AnyViewCallable`).

Reply:
OK I started doing that but then realized that I was bringing over half of that file since there are so many types that reference each other. I'm not confident I can do it properly so I'll leave a note to update this annotation once those type definitions are ported over in a different PR.

Reply:
Ok, no problem.